cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -78,18 +80,167 @@ def oid_as_pk_is_deprecated(cls, v):
+
+        ret = True
+
+        def inject(element: str):
+            return f"{namespace}__{element}"
+
+        # Determine whether we need to inject our sObject.
+        if inject_namespaces and self._is_injectable(self.sf_object):
+            global_describe = {
+                entry["name"]: entry
+                for entry in org_config.salesforce_client.global_describe()["sobjects"]
+            }
+
+            if (
+                self.sf_object in global_describe
+                and inject(self.sf_object) in global_describe
+            ):
+                logger.warning(
+                    f"Both {self.sf_object} and {inject(self.sf_object)} are present in the target org. Using {self.sf_object}."
+                )
+
+            if (
+                self.sf_object not in global_describe
+                and inject(self.sf_object) in global_describe
+            ):
+                self.sf_object = inject(self.sf_object)
+
+        # Validate our access to this sObject.
+        if self.sf_object not in global_describe or (
+            not global_describe[self.sf_object]["createable"]
+            and operation is DataOperationType.INSERT
+        ):
+            logger.warning(
+                f"sObject {self.sf_object} is not present or does not have the correct permissions."
+            )
+            ret = False
+
+        if not ret:
+            # Don't attempt to validate field permissions if the object doesn't exist.
+            return ret
+
+        # Validate, inject, and drop (if configured) fields.
+        describe = getattr(org_config.salesforce_client, self.sf_object).describe()
+        describe = {entry["name"]: entry for entry in describe["fields"]}
+
+        orig_fields = self.fields.copy()
+        for f, db in orig_fields.items():
+            # Do we need to inject this field?
+            if inject_namespaces and self._is_injectable(f):
+                if f in describe and inject(self.sf_object) in describe:
+                    logger.warning(
+                        f"Both {self.sf_object}.{f} and {self.sf_object}.{inject(f)} are present in the target org. Using {f}."
+                    )
+
+                if f not in describe and inject(f) in describe:
+                    self.fields[inject(f)] = db
+                    del self.fields[f]
+                    f = inject(f)
+
+            # Do we have the right permissions for this field, or do we need to drop it?
+            if f not in describe or (
+                not describe[f]["createable"] and operation is DataOperationType.LOAD
+            ):
+                logger.warning(
+                    f"Field {self.sf_object}.{f} is not present or does not have the correct permissions."
+                )
+                if drop_missing:
+                    del self.fields[f]
+                else:
+                    ret = False
+
+        # Validate, inject, and drop (if configured) lookups.
```

Comment:
So..close to the above. Could it be factored into a method and take the dictionary to work with as a param?

Reply:
Yep. Apologies, I failed to push a revision this morning that factors that out.